### PR TITLE
feat: api action endpoint scaffolding (types, db) #1077

### PR DESCRIPTION
**Pull request by ClaireMao1996**
Close https://github.com/samatechtw/pubstudio/issues/1077

### CODE REVIEW

File: backend/site-api/src/app/custom/custom_data.rs [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    println!("{}", dto.data);
+
+    return match dto.action {
+        Action::CreateTable => {
+            let create_table: CreateTable = parse_request_data(dto.data)?;
+
+            // TODO: Do something with create table
+            /*
+            {
+              "action": "CreateTable",
```
Comment (ClaireMao1996) [inline backend/site-api/src/app/custom/custom_data.rs:51]: I've attached example DTO within the code for testing.

**Comment by ClaireMao1996** [comment]
@sampullman ,

This PR is to establish the framework first, making it easier for us to directly implement the details of each action later on.

Let me know where adjustments are needed. Thanks!

**Comment by sampullman** [comment]
Got it! I'll review this, then rebase on a new feature branch called `feature-site-api-customization`. Since the frontend parts won't make sense until the whole feature is ready, I want to make sure we can still push bug fixes to production without releasing part of the feature.

**Comment by ClaireMao1996** [comment]
Got it! That makes sense! I just created the branch `feature-site-api-customization` from main, and I've re-pointed this PR to it.

File: backend/site-api/src/db/site_db_pool_manager.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+use lib_shared_site_api::db::db_error::DbError;
+use sqlx::SqlitePool;
+use tokio::sync::RwLock;
+
+pub type SiteDbPools = Arc<RwLock<HashMap<String, SqlitePool>>>;
+
+#[derive(Clone)]
+pub struct DbPoolManager {
+    pub site_db_pools: SiteDbPools,
```
[the remark this answers is not in the document]
Comment (sampullman) [reply backend/site-api/src/db/site_db_pool_manager.rs:11]: Makes sense!

**Comment by ClaireMao1996** [comment]
I'm sorry for making further adjustments, but I created each file first.

**Comment by sampullman** [comment]
No problem, I just double checked everything at it looks fine to me!

**Pull request merged**